### PR TITLE
remove 'tye.yaml' verbiage for YAML errors (#1033)

### PR DESCRIPTION
**Pull request by brettveenstra**
leverage file info when available, otherwise use more generic 'YAML file' phrasing

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=1034) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=1034) before we can accept your contribution.<br/><br/>:x: brettveenstra [sign now](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=1034)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/tye?pullRequest=1034) it.</sub>

File: src/Microsoft.Tye.Core/CoreStrings.resx [hunk 1]
```diff
@@ -166,13 +166,13 @@
     <value>Services must have unique names.</value>
   </data>
   <data name="UnexpectedType" xml:space="preserve">
-    <value>Unexpected node type in tye.yaml. Expected "{expected}" but got "{actual}".</value>
+    <value>Unexpected node type in YAML file. Expected "{expected}" but got "{actual}".</value>
```
Comment (jkotalik) [inline src/Microsoft.Tye.Core/CoreStrings.resx:169]:
```suggestion
    <value>Unexpected node type in tye configuration. Expected "{expected}" but got "{actual}".</value>
```

Comment (jkotalik) [reply src/Microsoft.Tye.Core/CoreStrings.resx:169]: And all other places.

File: src/Microsoft.Tye.Core/CoreStrings.resx [hunk 2]
```diff
@@ -166,13 +166,13 @@
...
   </data>
   <data name="UnrecognizedKey" xml:space="preserve">
-    <value>Unexpected key "{key}" in tye.yaml.</value>
+    <value>Unexpected key "{key}" in YAML file.</value>
```
Comment (davidfowl) [inline src/Microsoft.Tye.Core/CoreStrings.resx:172]:
```suggestion
    <value>Unexpected key "{key}" in the tye configuration file.</value>
```

File: src/Microsoft.Tye.Core/CoreStrings.resx [hunk 3]
```diff
@@ -166,13 +166,13 @@
...
   </data>
   <data name="UnexpectedTypes" xml:space="preserve">
-    <value>Unexpected node type in tye.yaml. Expected one of ({expected}) but got "{actual}".</value>
+    <value>Unexpected node type in YAML file. Expected one of ({expected}) but got "{actual}".</value>
```
Comment (davidfowl) [inline src/Microsoft.Tye.Core/CoreStrings.resx:175]:
```suggestion
    <value>Unexpected node type in the tye configuration file. Expected one of ({expected}) but got "{actual}".</value>
```

File: src/Microsoft.Tye.Core/CoreStrings.resx (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.Tye.Core/CoreStrings.resx:169]:
```suggestion
    <value>Unexpected node type in the tye configuration file. Expected "{expected}" but got "{actual}".</value>
```

File: test/UnitTests/TyeDeserializationValidationTests.cs [hunk 4]
```diff
@@ -274,6 +276,24 @@ public void LivenessProbeSuccessThresholdMustBeOne()
...
+
+            using var parser = new YamlParser(input, new FileInfo("foobar.yml"));
+            try
+            {
+                parser.ParseConfigApplication();
+                Assert.False(true, "YML parsing exception expected with supplied input");
+            }
+            catch (TyeYamlException e)
+            {
+                Assert.StartsWith("Error parsing 'foobar.yml': (2, 1): Unexpected node type in YAML file.", e.Message);
```
Comment (davidfowl) [inline test/UnitTests/TyeDeserializationValidationTests.cs:293]:
```suggestion
                Assert.StartsWith("Error parsing 'foobar.yml': (2, 1): Unexpected node type in the tye configuration file.", e.Message);
```

File: src/Microsoft.Tye.Core/CoreStrings.resx (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.Tye.Core/CoreStrings.resx:169]:
```suggestion
    <value>Unexpected node type in the tye configuration file. Expected "{expected}" but got "{actual}".</value>
```

File: src/Microsoft.Tye.Core/Serialization/YamlParser.cs [hunk 5]
```diff
@@ -106,10 +111,15 @@ public static void ThrowIfNotYamlSequence(string key, YamlNode node)
             }
         }
 
-        public static void ThrowIfNotYamlMapping(YamlNode node)
+        public static void ThrowIfNotYamlMapping(YamlNode node, FileInfo? fileInfo = null)
         {
             if (node.NodeType != YamlNodeType.Mapping)
             {
+                if (fileInfo!= null)
```
Comment (davidfowl) [inline src/Microsoft.Tye.Core/Serialization/YamlParser.cs:118]:
```suggestion
                if (fileInfo != null)
```

**Review by davidfowl — APPROVED** [review]

**Pull request merged**